### PR TITLE
feat: Endpointref

### PR DESCRIPTION
**Pull request by brianointel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/master/.github/CONTRIBUTING.md.

## What is the current behavior?

added a unit test to check for noauth path; return noauth path when mac is invalid

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## New Imports
Are there any new imports or modules? If so, what are they used for and why?

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

**Review by ajcasagrande — CHANGES_REQUESTED** [review]

**Review by ChristianDarr-personal — APPROVED** [review]
LGTM

File: internal/driver/credentials_test.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+package driver
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestTryGetCredentials verifies when a credential is passed with a noAuth path, noAuth credentials are returned.
```
Comment (ajcasagrande) [inline internal/driver/credentials_test.go:16]: Fix function name in comment

**Review by ajcasagrande — CHANGES_REQUESTED** [review]
Last thing

**Review by ajcasagrande — APPROVED** [review]
LGTM

**Pull request merged**